requirements.txt

```diff
@@ -9,4 +9,5 @@ ipaddress>=1.0.17;python_version=="2.7"  # PSF
 websocket-client>=0.32.0,!=0.40.0,!=0.41.*,!=0.42.* # LGPLv2+
 requests # Apache-2.0
 requests-oauthlib # ISC
-adal>=1.0.2 # MIT
+adal>=1.0.2;python_version=='2.7' # MIT
+adal>=1.0.2;python_version=='3' # MIT
```

Comment:
I am afraid this won’t work - see Travis CI information. Supported environment markers are documented in PEP 496 (
https://www.python.org/dev/peps/pep-0496/). 

To test your changes, you’ll need to have a Python 2.7 and a python 3.4, 3.5, 3.6 environment (as specified in Travis CI, but usually one of 3.4/5/6 is enough). Use setuptools to install your package from your local source, and use ```python -m unittest discover``` to test your code. Changing adal to be optional will at least require restructuring of existing code to let it be allowed as an extra install component and will break existing user experience using Azure. I suggest you consult the owners of the repo on how they want to do it.

Reply:
@micw523 I'm testing the same way you said. 
Adding this in requirements.txt
```
adal>=1.0.2;python_version=='2.7' # MIT
adal>=1.0.2;python_version=='3.5' # MIT
adal>=1.0.2;python_version=='3.6' # MIT
adal>=1.0.2;python_version=='3.7' # MIT```

All the test are passing locally but I am not sure if this is a good way to do.
Let me talk to the maintainer of this project.

Reply:
You probably have a local copy of adal. You can try using ```pip``` to list your installed packages. Try start with a fresh python environment and install from your local source with setuptools.

I think you probably has misunderstood me - I was simply saying ```python_version='3' ``` is not a valid environment marker. You'll need to restructure the code in kubernetes/python-base. Most likely there will be only a one-line change in the requirements.txt.